src/mlpack/core/data/load_image_impl.hpp

```diff
@@ -0,0 +1,217 @@
+
+// In case it hasn't been included yet.
+#include "load_image.hpp"
+
+namespace mlpack {
+namespace data {
+
+const std::vector<std::string> LoadImage::fileTypes({"jpg", "png", "tga",
+    "bmp", "psd", "gif", "hdr", "pic", "pnm"});
+
+LoadImage::LoadImage():
+        matrixWidth(0),
+        matrixHeight(0),
+        channels(3)
+{
+
+}
+
+LoadImage::LoadImage(const size_t width,
+          const size_t height,
+          const size_t channels):
+          matrixWidth(width),
+          matrixHeight(height),
+          channels(channels)
+{
+
+}
+
+LoadImage::~LoadImage()
+{
+  // Do nothing.
+}
+
+bool LoadImage::ImageFormatSupported(const std::string& fileName)
+{
+  // Iterate over all supported file types.
+  for (auto extension : fileTypes)
+    if (extension == Extension(fileName))
+      return true;
+  return false;
+}
+
+bool LoadImage::Load(const std::string& fileName,
+                      bool flipVertical,
+                      arma::Mat<unsigned char>&& outputMatrix,
+                      size_t &width,
+                      size_t &height,
+                      size_t &channels)
+{
+  unsigned char *image;
+
+  if (!ImageFormatSupported(fileName))
+  {
+    std::ostringstream oss;
+    oss << "File type " << Extension(fileName) << " not supported.\n";
+    oss << "Currently it supports ";
+    for (auto extension : fileTypes)
+      oss << " " << extension;
+    oss << std::endl;
+    throw std::runtime_error(oss.str());
+    return false;
+  }
+
+  stbi_set_flip_vertically_on_load(flipVertical);
+
+  // Temporary variablesneeded as stb_image.h supports int parameters.
+  int tempWidth, tempHeight, tempChannels;
+
+  // For grayscale images
+  if (channels == 1)
+  {
+    image = stbi_load(fileName.c_str(),
+             &tempWidth,
+             &tempHeight,
+             &tempChannels,
+             STBI_grey);
+  }
+  else
+  {
+    image = stbi_load(fileName.c_str(),
+             &tempWidth,
+             &tempHeight,
+             &tempChannels,
+             STBI_rgb);
+  }
+
+  if (tempWidth <= 0 || tempHeight <= 0)
+  {
+    std::ostringstream oss;
+    oss << "Image '" << fileName << "' not found." << std::endl;
+    free(image);
+    throw std::runtime_error(oss.str());
+    return false;
+  }
+  width = tempWidth;
+  height = tempHeight;
+  channels = tempChannels;
+  size_t size = width * height * channels;
+
+  // Copy image into armadillo Mat.
+  outputMatrix = arma::Mat<unsigned char>(image, 1, size, true, true);
+
+  // Free the image pointer.
+  free(image);
+  return true;
+}
+
+bool LoadImage::Load(const std::string& fileName,
+                    bool flipVertical,
+                    arma::Mat<unsigned char>&& outputMatrix)
+{
+  size_t width, height;
+  bool status = Load(fileName,
+      flipVertical,
+      std::move(outputMatrix),
+      width, height, channels);
+  if (!status)
+    return status;
+
+  Log::Info << width <<" "<< height <<" "<< channels << std::endl;
```

Comment:
Do you think it would be helpful to add some more information here something like:

```cpp
Log::Info << "Image width: " << width <<"  height: "<< height <<" channels: " << channels << std::endl;
```